pageserver/src/walredo.rs

```diff
@@ -626,24 +626,20 @@ impl PostgresRedoProcess {
 
         // Create empty data directory for wal-redo postgres, deleting old one first.
         if datadir.exists() {
-            info!(
-                "old temporary datadir {} exists, removing",
-                datadir.display()
-            );
-            fs::remove_dir_all(&datadir)?;
+            info!("old temporary datadir {datadir:?} exists, removing");
+            fs::remove_dir_all(&datadir).map_err(|e| {
+                Error::new(
+                    e.kind(),
+                    format!("Failed to remove old temporary dir {datadir:?}: {e}"),
+                )
+            })?;
         }
-        let pg_bin_dir_path = conf.pg_bin_dir(pg_version).map_err(|e| {
-            Error::new(
-                ErrorKind::Other,
-                format!("incorrect pg_bin_dir path: {}", e),
-            )
-        })?;
-        let pg_lib_dir_path = conf.pg_lib_dir(pg_version).map_err(|e| {
-            Error::new(
-                ErrorKind::Other,
-                format!("incorrect pg_lib_dir path: {}", e),
-            )
-        })?;
+        let pg_bin_dir_path = conf
+            .pg_bin_dir(pg_version)
+            .map_err(|e| Error::new(ErrorKind::Other, format!("incorrect pg_bin_dir path: {e}")))?;
+        let pg_lib_dir_path = conf
+            .pg_lib_dir(pg_version)
+            .map_err(|e| Error::new(ErrorKind::Other, format!("incorrect pg_lib_dir path: {e}")))?;
```

Comment:
```suggestion
            .map_err(|e| Error::new(e.kind(), format!("incorrect pg_bin_dir path: {e}")))?;
        let pg_lib_dir_path = conf
            .pg_lib_dir(pg_version)
            .map_err(|e| Error::new(e.kind(), format!("incorrect pg_lib_dir path: {e}")))?;
```

Reply:
I wish I could apply this (multiline edits are there? nice), but `e` is of type `anyhow::Error` for both of these hence `.kind()` does not exist there.

https://github.com/neondatabase/neon/blob/7920b39a275a7bd13155d8726c92cd417bf7e2f9/pageserver/src/config.rs#L514-L527